src/commands/createStaticWebApp/GitHubBranchListStep.ts

```diff
@@ -3,23 +3,28 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { Octokit, RestEndpointMethodTypes } from '@octokit/rest';
 import { AzureWizardPromptStep, IAzureQuickPickItem } from 'vscode-azureextensionui';
-import { githubApiEndpoint } from '../../constants';
 import { ext } from '../../extensionVariables';
-import { createGitHubRequestOptions, getGitHubQuickPicksWithLoadMore, getGitHubTree, getRepoFullname, gitHubBranchData, gitHubWebResource, ICachedQuickPicks } from '../../utils/gitHubUtils';
+import { BranchData } from '../../gitHubTypings';
+import { getGitHubQuickPicksWithLoadMore, getGitHubTree, getRepoFullname, ICachedQuickPicks } from '../../utils/gitHubUtils';
 import { localize } from '../../utils/localize';
 import { nonNullProp, nonNullValueAndProp } from '../../utils/nonNull';
+import { createOctokitClient } from '../github/createOctokitClient';
 import { IStaticWebAppWizardContext } from './IStaticWebAppWizardContext';
 
+type ReposListBranchesParameters = RestEndpointMethodTypes['repos']['listBranches']['parameters'];
+
 export class GitHubBranchListStep extends AzureWizardPromptStep<IStaticWebAppWizardContext> {
     public async prompt(context: IStaticWebAppWizardContext): Promise<void> {
-        let branchData: gitHubBranchData | undefined;
+        let branchData: BranchData | undefined;
         const { owner, name } = getRepoFullname(nonNullProp(context, 'repoHtmlUrl'));
         const placeHolder: string = localize('chooseBranch', 'Choose branch for repository "{0}/{1}"', owner, name);
-        const requestOption: gitHubWebResource = await createGitHubRequestOptions(context.accessToken, `${githubApiEndpoint}/repos/${owner}/${name}/branches`);
```

Comment:
We use `context.accessToken` here because we want to use the same token for the whole wizard instead of re-getting a token each time (which could affect performance and/or prompt the user for password unless they pick "Always Allow"). Can you find a way to replicate that with the new octokit stuff? A few options:
1. Add an optional 'token' param to `createOctokitClient` that would bypass `getGitHubAccessToken`
1. Add the client itself to the wizard context